code-challenges.test.js

```diff
@@ -15,42 +16,132 @@
+    for (let i = 2; i < number; i++) {
+      var1.push(var1[i-1] + var1[i-2]);
+    }
+    return var1
+}
+
+
+// PASS  ./code-challenges.test.js
+//   fibonaciArray
+//     ✓ takes in a number (greater than 2) and returns an array that length containing the numbers of the Fibonacci sequence. (1 ms)
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
 
 
 // --------------------2) Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
-// Expected output: [-9, 7, 9, 199]
-
-const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
-// Expected output: [-823, 7, 23]
 
+describe ("newArray", () => {
+    it ("takes in an array and returns a new array of only odd numbers sorted from least to greatest.", () => {
+        const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
+        const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
+      expect(newArray(fullArr1)).toEqual([-9, 7, 9, 199])
+      expect(newArray(fullArr2)).toEqual([-823, 7, 23])
+    })
+  })
 
-// b) Create the function that makes the test pass.
+// Succesful failure 
+// ● newArray › takes in an array and returns a new array of only odd numbers sorted from least to greatest.
+// ReferenceError: newArray is not defined
 
 
-// --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+// b) Create the function that makes the test pass.
+// Pesudo Code 
+// create a function: oddNumbers 
+// input: array given 
+// returns a new array with only odd numbers sorted from least to greatest 
+// will need to use .filter() to return a new array with only odd numbers 
+// our returned odd numbers need to be ordered from least to greatest -->use .sort()
+
+
+const oddNumbers = (array) => {
+    let odds = []
+    array.filter(value => value % 2 !== 0)
+    return odds.filter(value  => typeof value === 'numbers')
+}
+// console.log(oddNumbers(fullArr1));
+
+// I got stuck on the code above and could not figure out how to create the function that would return an array of only odd numbers sorted from least to greatest. I got my code to return all the odd values initially but the output was also including strings and numbers that it recognized as odd values. I then attempted to add the typeof method to return only the odd numbers but for some reason my code is not running, and I want to say it is due to the syntax and structure of my function? I also created the code below as a reference to how to obtain only numbers in our array but could not figure out where else to go with my code in order to get the result the question is asking. Would a for loop be a better approach here? 
```

Comment:
Thank you for explaining your approach. Let me know if my explanation made any sense.